build/longhaul.vsts-ci.yml

```diff
@@ -19,11 +19,10 @@ jobs:
       sudo cat /etc/os-release
       sudo apt show valgrind
       sudo chmod --recursive ugo+rw cmake
-      sudo chmod 755 jenkins/longhaul_c.sh
       sudo -E ./jenkins/longhaul_c.sh
     displayName: 'Build'
   - script: |
-      cd cmake && sudo -E ../build_all/linux/run_tests.sh
+      cd cmake && sudo -E ../build_all/linux/run_tests_memcheck.sh
```

Comment:
For the E2E tests we generate separate ctest depending on the build options and run_tests is able to run valgrind with all options as well as all the overrides that several of the dependencies require.

Why does this need to be different?

Reply:
because the E2E tests uses valgrind settings that would be overridden by setting -T MemCheck